src/catalog/database_catalog.cpp

```diff
@@ -136,17 +140,52 @@ std::shared_ptr<TableCatalogObject> DatabaseCatalogObject::GetTableObject(
  * @return  Shared pointer to the requested table catalog object
  */
 std::shared_ptr<TableCatalogObject> DatabaseCatalogObject::GetTableObject(
-    const std::string &table_name, bool cached_only) {
-  auto it = table_name_cache.find(table_name);
+    const std::string &table_name, const std::string &session_namespace,
+    const std::string &table_namespace, bool cached_only) {
+  // no namespace specified.
+  if (table_namespace.empty()) {
+    // search session first.
+    std::string table_key = session_namespace + std::string("#") + table_name;
```

Comment:
It seems like the logic from 148-168 has some overlaps. Create a function might improve readability/